addons/addon-base/packages/services/lib/s3-service.js

```diff
@@ -177,13 +177,37 @@ class S3Service extends Service {
       .promise();
   }
 
-  async createFolder(bucketName, folderName) {
+  async createPath(bucketName, folderName) {
     const params = {
       Bucket: bucketName,
       Key: folderName,
     };
     return this.api.putObject(params).promise();
   }
+
+  async clearPath(bucketName, dir) {
+    const listParams = {
+      Bucket: bucketName,
+      Prefix: dir,
+    };
+
+    const listedObjects = await this.api.listObjectsV2(listParams).promise();
+
+    if (listedObjects.Contents.length === 0) return;
+
+    const deleteParams = {
+      Bucket: bucketName,
+      Delete: { Objects: [] },
+    };
+
+    listedObjects.Contents.forEach(({ Key }) => {
+      deleteParams.Delete.Objects.push({ Key });
+    });
+
+    await this.api.deleteObjects(deleteParams).promise();
```

Comment:
Do we need to verify the response of deleteObjects API and verify that there were no errors? Also, if the bucket is versioned (which I thought was the case by default), what is the behavior we want? Do we want to mark the objects for deletion or actually delete them?

Reply:
Since the bucket is enabled with versioning. Only delete marker will be put on objects.
Test added in commit: 8eccd37803ee539e96dbc9839eab3bae57be8d5a